x-pack/plugins/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -174,11 +183,15 @@ export class AlertsClient<
 
   public async search(queryBody: SearchRequest['body']): Promise<SearchResult<AlertData>> {
     const esClient = await this.options.elasticsearchClientPromise;
+    const index = this.isUsingDataStreams()
+      ? this.indexTemplateAndPattern.alias
+      : this.indexTemplateAndPattern.pattern;
     const {
       hits: { hits, total },
     } = await esClient.search<Alert & AlertData>({
-      index: this.indexTemplateAndPattern.pattern,
+      index,
       body: queryBody,
+      ignore_unavailable: true,
```

Comment:
A new thing!  While comparing ruleRegistry to this code, I noticed they were doing `ignore_unavailable: true`, which seems like it makes sense for us as well.  Basically, drive-by.  No problem removing, but seems useful.